e2e/etcd_test.go

```diff
@@ -401,28 +414,32 @@ func (epc *etcdProcessCluster) Close() error {
 	return err
 }
 
-func (ep *etcdProcess) Restart() error {
+func (ep *etcdProcess) Restart(unix bool) error {
 	newEp, err := newEtcdProcess(ep.cfg)
 	if err != nil {
-		ep.Stop()
+		ep.Stop(unix)
 		return err
 	}
 	*ep = *newEp
 	if err = ep.waitReady(); err != nil {
-		ep.Stop()
+		ep.Stop(unix)
 		return err
 	}
 	return nil
 }
 
-func (ep *etcdProcess) Stop() error {
+func (ep *etcdProcess) Stop(unix bool) error {
 	if ep == nil {
 		return nil
 	}
 	if err := ep.proc.Stop(); err != nil {
 		return err
 	}
 	<-ep.donec
+
+	if unix {
+		os.RemoveAll(ep.cfg.purl.Host)
```

Comment:
I think the unix listener already does this by default? https://github.com/coreos/etcd/blob/master/pkg/transport/unix_listener.go#L25